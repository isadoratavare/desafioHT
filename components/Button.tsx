import React from "react";
import {
    TouchableOpacity,
    Text,
    StyleSheet,
    useColorScheme,
} from "react-native";

type ButtonProps = {
    title: string;
    onPress?: () => void;
};

const Button = ({ title, onPress }: ButtonProps) => {
    const colorScheme = useColorScheme();
    const color = colorScheme === "dark" ? "#5566FF" : "";

    return (
        <TouchableOpacity
            onPress={onPress}
            style={[styles.buttonContainer, { backgroundColor: color }]}
        >
            <Text style={styles.buttonText}>{title}</Text>
        </TouchableOpacity>
    );
};

const styles = StyleSheet.create({
    buttonContainer: {
        width: "100%",
        alignItems: "center",
        marginVertical: 5,
        paddingHorizontal: 15,
        borderRadius: 15,
        paddingVertical: 20,
    },
    buttonText: {
        color: "white",
        fontFamily: "Lato-Bold",
        fontWeight: "400",
    },
});

export default Button;
