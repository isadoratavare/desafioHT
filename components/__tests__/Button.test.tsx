import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Button from '../Button';

describe('Button component', () => {
  it('renders correctly with title', () => {
    const { getByText } = render(<Button title="Press Me" />);
    const buttonElement = getByText('Press Me');
    expect(buttonElement).toBeTruthy();
  });

  it('executes onPress callback when pressed', () => {
    const mockOnPress = jest.fn();
    const { getByText } = render(<Button title="Press Me" onPress={mockOnPress} />);
    const buttonElement = getByText('Press Me');
    fireEvent.press(buttonElement);
    expect(mockOnPress).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<Button title="Press Me" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
