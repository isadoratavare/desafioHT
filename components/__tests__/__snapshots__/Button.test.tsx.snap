// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#5566FF",
      "borderRadius": 15,
      "marginVertical": 5,
      "opacity": 1,
      "paddingHorizontal": 15,
      "paddingVertical": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontFamily": "Lato-Bold",
        "fontWeight": "400",
      }
    }
  >
    Press Me
  </Text>
</View>
`;
